source/common/network/matching/inputs.h

```diff
@@ -238,6 +241,65 @@ class ApplicationProtocolInputFactory
   ApplicationProtocolInputFactory() : BaseFactory("application_protocol") {}
 };
 
+template <class MatchingDataType> class UriSanInput : public Matcher::DataInput<MatchingDataType> {
```

Comment:
These look good to me, though I think you need the factory registration bit? A test verifying that the registration is complete would be nice as well

Reply:
Done.

BTW, do you think it is a good idea to separate them from network inputs and move to something like SSL inputs since they are TLS specific?

Reply:
Yes, I do think it would make sense to move them to be SSL inputs.